src/runtime/schema/settings.ts

```diff
@@ -81,104 +134,39 @@ export function changeSettings(state: SettingsData, newSettings: SettingsInput):
   }
 
   if (newSettings.connections) {
-    state.connections = state.connections ?? {}
-    const { types, ...connectionPluginConfig } = newSettings.connections
-    if (types) {
-      state.connections.types = state.connections.types ?? {}
-      Object.assign(state.connections.types, types)
-    }
-    Object.assign(state.connections, connectionPluginConfig)
+    // todo deep merge
+    Object.assign(state.connections, {
+      ...newSettings.connections,
+      nexusFieldName: CONNECTION_FIELD_BUILDER_NAME,
+    })
   }
 }
 
-export function mapSettingsToNexusSchemaConfig(
-  frameworkPlugins: Plugin.RuntimeContributions[],
-  settings: SettingsData
-): NexusSchemaConfig {
-  const runtimeTypegenConfig = {
-    // Always false here, then set to true in the reflection module
-    outputs: false,
-    shouldGenerateArtifacts: false,
-    shouldExitAfterGenerateArtifacts: false,
-  }
-
-  const baseConfig: NexusSchemaConfig = {
-    nonNullDefaults: {
-      input: !(settings?.nullable?.inputs ?? true),
-      output: !(settings?.nullable?.outputs ?? true),
+/**
+ * Get the default settings.
+ */
+function defaultSettings(): SettingsData {
```

Comment:
Looks small but this was a pretty fundamental change to our codebase. Some defaults that lived very far from this source code like `rootTypingsGlobPattern` are now here.